extmod/modubinascii.c

```diff
@@ -89,14 +88,89 @@ mp_obj_t mod_binascii_unhexlify(mp_obj_t data) {
 }
 MP_DEFINE_CONST_FUN_OBJ_1(mod_binascii_unhexlify_obj, mod_binascii_unhexlify);
 
+static const char *base64codes = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+
+mp_obj_t mod_binascii_a2b_base64(mp_obj_t data) {
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(data, &bufinfo, MP_BUFFER_READ);
+    if (bufinfo.len % 4 != 0) {
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ValueError, "incorrect padding"));
```

Comment:
mp_obj_new_exception_msg_varg() should probably be avoided here (use mp_obj_new_exception_msg())